### DISCUSSION TITLE
Not able to import modules in versions after 0.1.13rc0 (f79282c4)

### DISCUSSION THREAD

**Discussion by temptemp3** [reach-sh/reach-lang#1569] [General]
Good day.

I am having issues compiling a reach program using modules in v0.1.13-rc.0 are later. I started with the latest version to date 0.1.13-rc.3 (88e48902) and worked my way back to the latest stable version 0.1.12. Reach version 0.1.12 works as expect.

Here is what I am seeing on 0.1.13-rc.0 or later:

1) Compiling

Command: `./reach compile --install-pkgs`

Output:

REACH_VERSION: f79282c4
reachc: git: readCreateProcessWithExitCode: exec: invalid argument (Bad file descriptor)
reachc: error[RI0003]: Cannot install packages; did you mean to run with `--install-pkgs`?

  ./index.rsh:4:30:import from

  4| import { api, view, events } from "@temptemp3/arc-72#arc-72-v0.1.13r0:index.rsh";

2) Compiling with packages

Command: `./reach compile --install-pkgs`

Output: 

reachc: git: readCreateProcessWithExitCode: exec: invalid argument (Bad file descriptor)

2) Updating

Command `./reach update `

Output:

```
Please stand-by...reach: HttpExceptionRequest Request {
  host                 = "docs.reach.sh"
  port                 = 443
  secure               = True
  requestHeaders       = []
  path                 = "/reach"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
  proxySecureMode      = ProxySecureWithConnect
}
 (InternalException (HandshakeFailed (Error_Protocol ("certificate has unknown CA",True,UnknownCa))))
 ```

Here is the source code for the reach program. Not that I added an import line to an existing module.

https://github.com/temptemp3/arc-72/blob/discussions/1569/index.rsh

**Comment by jeapostrophe** [comment]
What platform are you on? One hypothesis is that you're on an Apple Silicon chip but some how are running x86 images.

**Reply by temptemp3** [reply]
Yes. I am on am on Apple Silicon. But I have been for some time with no issues. Everything works okay up to v0.1.12. However, I have had no success on any of the v0.1.13 rc versions.

**Reply by temptemp3** [reply]
Is there something *new* I might be missing in case of Apple Silicon?

**Reply by jeapostrophe** [reply]
Can you look at your Docker image info output?

**Reply by temptemp3** [reply]
The only thing that I noticed does not work is importing. So what I have done on this version is have the code available locally as a temporary measure.

**Reply by temptemp3** [reply]
It might be because of Rosetta that I am able to run x86 images.

**Reply by jeapostrophe** [reply]
Yes, that's what I think. If you were using the real ARM images then this problem would not happen

**Reply by temptemp3** [reply]
You are most likely right. How do I switch to the real ARM image?

**Reply by jeapostrophe** [reply]
I think the source of the problem is how you are running your terminal and/or Docker. I don't actually know how this works. You could open up Docker Desktop and kill machines?

**Reply by temptemp3** [reply]
I'll tinker with it some since docker on my chip is suspicious

**Reply by temptemp3** [reply]
How might I go about identifying the arm image?

**Reply by jeapostrophe** [reply]
Check out Docker Desktop's image list. If you look at this page --- https://hub.docker.com/r/reachsh/reach-cli/tags --- you'll see there are `linux/amd64` and `linux/arm64` versions. You want to see the ARM versions.

**Reply by temptemp3** [reply]
Got it. Thanks!

**Reply by temptemp3** [reply]
Might be because my hashes are off too
```
./reach hashes
reach-cli: f79282c4
reach: 88e48902
```

**Reply by temptemp3** [reply]
Not sure how helpful this is but in the latest image sudo appears to be missing 
```
+ package_install curl
+ command -v apt
+ PACKAGE_MANAGER=apt
+ command -v yum
+ sudo apt install curl
./reach: 165: sudo: not found
+ command -v make
```
Going to compare to the image of 0.1.12

**Reply by temptemp3** [reply]
Wasn't able to get anywhere with this but what I do know is that I am able to compile a reach (v0.1.13rc3) app on Apple Silicon that does not have import lines. Compiling reach programs for reach v0.1.12) doesn't have issues. I'm stuck so I'm just going to have to avoid using import for the time being and test again later.

**Reply by jeapostrophe** [reply]
I've verified that this is actually a problem by starting from scratch on my Apple Silicon machine. I'm going to convert to an issue and look into it more.

**Comment by temptemp3** [comment]
Great. Thanks!
